modules/apps/change-tracking/change-tracking-web/src/main/java/com/liferay/change/tracking/web/internal/servlet/taglib/ChangeTrackingIndicatorDynamicInclude.java

```diff
@@ -423,9 +429,14 @@ private Map<String, Object> _getReactData(
 
 		long ctCollectionId = CTConstants.CT_COLLECTION_ID_PRODUCTION;
 
-		String customProductionName = _getCTConfiguration(
-			themeDisplay.getCompanyId()
-		).customProductionName();
+		CTConfiguration ctConfiguration = _getCTConfiguration(
+			themeDisplay.getCompanyId());
+
+		LocalizedValuesMap localizedValuesMap =
+			ctConfiguration.customProductionName();
+
+		String customProductionName = localizedValuesMap.get(
+			themeDisplay.getLocale());
```

Comment:
Use `PublicationUtil.getCustomProductionName`

Reply:
There was a pre-existing `_getCTConfiguration(...)` method that defaults to a `_defaultCTConfiguration` (created in @Activate) if getting the CTConfiguration by companyId failed. Should that method be removed too and other usages replaced with `PublicationUtil.getCustomProductionName`?

Reply:
I see. Let's move the logic to PublicationUtil so that we have a consistent way for retrieving the configuration. 

Use an empty properties map to create the default configuration with default values 
```
ConfigurableUtil.createConfigurable(CTConfiguration.class, Collections.emptyMap());
```

We can remove the `configurationPid` component property and the `activate` method from ChangeTrackingIndicatorDynamicInclude. This way is only used to read **SYSTEM** scoped configurations. The configurationPid for scoped config looks like this `com.liferay.change.tracking.web.internal.configuration.CTConfiguration.scoped~2df7c11b-b4b2-4af1-8984-81c6bdd400b0` and can only be read through the ConfigurationProvider.